tests/Console/ConfigurationResolverTest.php

```diff
@@ -868,6 +869,123 @@ public function testResolveRulesWithOption()
         );
     }
 
+    /**
+     * @param string[] $rules
+     * @param string[] $negativeRules
+     *
+     * @dataProvider provideSubtractionCases
+     */
+    public function testResolveRulesWithNegativeSet(array $rules, array $negativeRules)
```

Comment:
1. maybe possible to not rely on real seats, but register mocked ones (with only few rules) ?
2. even moving "semi-resolving" from test method to data provider would make test method getting raw `input, expected` params, test method should avoid having code with logic